### PR TITLE
Catlike Grace works with Freerunning

### LINKED ISSUE #82818: Felinids with freerunning no longer land on their feet.
## Issue Summary
Felinids with freerunning do not land on their feet. After speaking with a maintainer it appears the code is not checking for both freerunning and catlike grace, overriding behavior.

## Round ID:
[227616](https://scrubby.melonmesa.com/round/227616)

## Reproduction:
As a felinid with the freerunning quirk fall one z level. Observe that you do not land on your feet.
See instead that you brace for the fall. You make a hard landing on the floor, but remain unharmed.

### PR DESCRIPTION
**Pull request by necromanceranne**
## About The Pull Request

If you have Catlike Grace when Freerunning takes effect, you do not suffer a knockdown. fixes https://github.com/tgstation/tgstation/issues/82818

## Why It's Good For The Game

Taking a quirk to do something worse than if you didn't have it at all is probably not the best outcome. Some may argue that not taking damage from a fall is actually a good thing, but in my experience the catlike grace bonus is a much better outcome.

If for whatever reason you now have both, they're no longer at odds.

Obviously, Freerunning only cares about dropping from a single level. If you take a tumble down into the depths of the ice moon, you're still landing on your feet and taking a ridiculous, bone shattering amount of damage.

Also I asked @MrMelbert if the PR was okay and he said 'so long as the PR is good' so hopefully this is up to snuff.

## Changelog
:cl:
fix: If you are a freerunner, you don't end up faceplanting despite having catlike grace.
/:cl:

### CODE REVIEW

**Comment by EricZilla** [comment]
Freerunning is really expensive so I think this is fine but HOO boy do I never wanna fight a felinid on tram after this

**Comment by Dawnseer** [comment]
> Freerunning is really expensive so I think this is fine but HOO boy do I never wanna fight a felinid on tram after this

Why fight them when you can just laser pointer em into the way of the tram when it arrives?

File: code/modules/mob/living/living.dm [hunk 1]
```diff
@@ -68,11 +68,13 @@
 	if(levels <= 1 && can_help_themselves)
 		var/obj/item/organ/external/wings/gliders = get_organ_by_type(/obj/item/organ/external/wings)
 		if(HAS_TRAIT(src, TRAIT_FREERUNNING) || gliders?.can_soften_fall()) // the power of parkour or wings allows falling short distances unscathed
+			var/graceful_landing = HAS_TRAIT(src, TRAIT_CATLIKE_GRACE) ? TRUE : FALSE
```
Comment (MrMelbert) [inline code/modules/mob/living/living.dm:71]:
```suggestion
			var/graceful_landing = HAS_TRAIT(src, TRAIT_CATLIKE_GRACE)
```

File: code/modules/mob/living/living.dm [hunk 2]
```diff
@@ -68,11 +68,13 @@
...
+			if(!graceful_landing)
+				Knockdown(levels * 4 SECONDS)
 			visible_message(
-				span_notice("[src] makes a hard landing on [impacted_turf] but remains unharmed from the fall."),
-				span_notice("You brace for the fall. You make a hard landing on [impacted_turf], but remain unharmed."),
+				span_notice("[src] makes a hard landing on [impacted_turf] but remains unharmed from the fall[graceful_landing ? " and stays on [p_their()] feet" : null]."),
+				span_notice("You brace for the fall. You make a hard landing on [impacted_turf], but remain unharmed[graceful_landing ? " and stay on your feet" : null]."),
```
Comment (MrMelbert) [inline code/modules/mob/living/living.dm:76]: If I can suggest a flavor change, non-graceful-landing should imply some kind of tuck-and-roll.

**Review by MrMelbert — CHANGES_REQUESTED** [review]
There should still be a penalty to falling, because the intent is for there to never be a consequence-free tumble. (Which is why this interaction was intended, IE not a bug) 

Should apply the slowdown (`/datum/movespeed_modifier/landed_on_feet`) for 4 seconds.

File: code/modules/mob/living/living.dm [hunk 3]
```diff
@@ -68,11 +68,15 @@
...
+				Knockdown(levels * 4 SECONDS)
 			visible_message(
-				span_notice("[src] makes a hard landing on [impacted_turf] but remains unharmed from the fall."),
-				span_notice("You brace for the fall. You make a hard landing on [impacted_turf], but remain unharmed."),
+				span_notice("[src] makes a hard landing on [impacted_turf] but remains unharmed from the fall[graceful_landing ? " and stays on [p_their()] feet" : " by tucking in rolling into the landing"]."),
+				span_notice("You brace for the fall. You make a hard landing on [impacted_turf], but remain unharmed[graceful_landing ? " while landing on your feet" : " by tucking in and rolling into the landing"]."),
 			)
-			Knockdown(levels * 4 SECONDS)
+			add_movespeed_modifier(/datum/movespeed_modifier/landed_on_feet)
+			addtimer(CALLBACK(src, TYPE_PROC_REF(/mob, remove_movespeed_modifier), /datum/movespeed_modifier/landed_on_feet), levels * 2 SECONDS)
```
Comment (MrMelbert) [inline code/modules/mob/living/living.dm:79]: You're currently adding the movespeed modifier regardless of whether it's a graceful landing or not. 

Also make it `* 3 seconds`

**Review by MrMelbert — CHANGES_REQUESTED** [review]

**Review by Ghommie — APPROVED** [review]

**Pull request merged**